SPECIFICATION.md

```diff
@@ -0,0 +1,93 @@
+<h1>Provider Specification</h1>
+
+[TOC]
+
+# General
+- **name** Any record name supplied to the provider must be the full hostname.
+- **ttl** Reasonable default is 6 hours since it's supported by most services. Any service that does not support this must be explicitly mentioned somewhere.
```

Comment:
Adding validation to lexicon has a couple of advantages:

- better usability, users know immediately if something is wrong
- better error messages

But there are also some disadvantages:

- our specified provider TTL range may be out-of-date with the actual API, will require a new version to be released. 
- having a lexicon specific TTL default breaks the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment). Now instead of letting the API determine the default TTL if none is provided, 
lexicon will send its own default, which may not be expected. 
- if we decide to go down the route of client validation, we should also consider provider specific validation for record types, record content length, record character sets, and more.

Reply:
I figured 6 hours is a reasonable value supported by most providers. If it does not make sense, we can just have the provider handle it internally. Though, having a single default for all providers will make the API more consistent.